shop/models/fields.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from distutils.version import LooseVersion
+import re
+import subprocess
+from django.db import connection
+
+POSTGRES_FLAG = False
+if str(connection.vendor) == 'postgresql':
+    POSTGRES_FLAG = True
+try:
+    if POSTGRES_FLAG:
+        import psycopg2
+
+        psycopg2_version = re.search('([0-9.]+)', psycopg2.__version__ or "0").group(0)
+        try:
+            process = subprocess.Popen(
```

Comment:
Never ever run shell command from within Django. This is very unportable and dangerous.
A better way to achieve this is to use raw SQL:
```
with connection.cursor() as cursor:
    cursor.execute("SELECT version()")
    row = cursor.fetchone()
```
``row`` then contains the version you can further process.